python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -286,56 +284,53 @@ def forward(self, g, edge_feat):
-            num_nodes = g.batch_num_nodes()[idx_g]
-            ubg = unbatch(g)[idx_g]
-            edata = edge_feat[
-                sum_num_edges: (sum_num_edges + g.batch_num_edges()[idx_g]), :
-            ]
-            sum_num_edges = sum_num_edges + g.batch_num_edges()[idx_g]
+        path_encoding = []
+
+        for ubg in ubg_list:
+            num_nodes = ubg.num_nodes()
+            num_edges = ubg.num_edges()
+            edata = edge_feat[sum_num_edges: (sum_num_edges + num_edges)]
+            sum_num_edges = sum_num_edges + num_edges
             edata = th.cat(
-                (edata, th.unsqueeze(th.zeros(self.feat_dim), 0)), 0
+                (edata, th.zeros(1, self.feat_dim).to(edata.device)),
+                dim=0
             )
-            path = shortest_dist(ubg, root=None, return_paths=True)[1]
+            _, path = shortest_dist(ubg, root=None, return_paths=True)
             path_len = min(self.max_len, path.size(dim=2))
 
-            # shape of shortest path is [n, n, l], n = num_nodes, l = path_len
+            # shape: [n, n, l], n = num_nodes, l = path_len
             shortest_path = path[:, :, 0: path_len]
-            # shape of path_data is [n, n, l, d], d = feat_dim
+            # shape: [n, n]
+            shortest_distance = shortest_dist(
+                ubg, root=None, return_paths=False
+            )
+            # shape: [n, n, l, d], d = feat_dim
             path_data = edata[shortest_path]
-            # shape of embedding_idx is [l, h], h = num_heads
+            # shape: [l, h], h = num_heads
             embedding_idx = th.reshape(
                 th.arange(self.num_heads * path_len),
                 (path_len, self.num_heads)
             )
-            # shape of edge_embedding is [d, l, h]
+            # shape: [d, l, h]
             edge_embedding = th.permute(
                 self.embedding_table(embedding_idx), (2, 0, 1)
             )
 
             # [n, n, l, d] einsum [d, l, h] -> [n, n, h]
             # [n, n, h] -> [N, N, h], N = max_num_nodes, padded with -inf
-            if not idx_g:
-                path_encoding = th.full(
-                    (max_num_nodes, max_num_nodes, self.num_heads),
-                    float('-inf')
-                )
-                path_encoding[0: num_nodes, 0: num_nodes, :] = th.einsum(
-                    'xyld,dlh->xyh', path_data, edge_embedding
-                )
-                path_encoding = th.unsqueeze(path_encoding, 0)
-            else:
-                sub_path_encoding = th.full(
-                    (max_num_nodes, max_num_nodes, self.num_heads),
-                    float('-inf')
-                )
-                sub_path_encoding[0: num_nodes, 0: num_nodes, :] = th.einsum(
+            sub_encoding = th.full(
+                (max_num_nodes, max_num_nodes, self.num_heads),
+                float('-inf')
+            )
+            sub_encoding[0: num_nodes, 0: num_nodes] = th.div(

```

Comment:
`th.div` is something you forgot in the previous implementation?

Reply:
Not forgot to, but ignored. It is divided by constants, so I ignored it. But I think it is better to be faithful to original paper, so I added it yesterday.

Reply:
`shortest_distance` should be clamped, according to the upper bound `max_len`, I will add it later.